# week1  

# ANOVA ON GAPMINDER 

## importing data

In [105]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import statsmodels.formula.api as smf

In [47]:
import statsmodels.stats.multicomp as multi

In [48]:
data = pd.read_csv("gapminder.csv")

In [49]:
data.head()

country   incomeperperson alcconsumption armedforcesrate  \
0  Afghanistan                              .03        .5696534   
1      Albania  1914.99655094922           7.29       1.0247361   
2      Algeria  2231.99333515006            .69        2.306817   
3      Andorra  21943.3398976022          10.17                   
4       Angola  1381.00426770244           5.57       1.4613288   

  breastcancerper100th      co2emissions  femaleemployrate hivrate  \
0                 26.8          75944000  25.6000003814697           
1                 57.4  223747333.333333  42.0999984741211           
2                 23.5  2932108666.66667  31.7000007629394      .1   
3                                                                    
4                 23.1         248358000  69.4000015258789       2   

    internetuserate lifeexpectancy     oilperperson polityscore  \
0  3.65412162280064         48.673                            0   
1  44.9899469578783         76.918                            9   
2  12.5000733055148         73.131  .42009452521537           2   
3                81                                               
4  9.99995388324075         51.093                           -2   

  relectricperperson   suicideper100th        employrate urbanrate  
0                     6.68438529968262  55.7000007629394     24.04  
1   636.341383366604  7.69932985305786  51.4000015258789     46.72  
2   590.509814347428   4.8487696647644              50.5     65.22  
3                     5.36217880249023                       88.92  
4   172.999227388199  14.5546770095825  75.6999969482422      56.7

In [81]:
data1 = data.iloc[:,[0,1,2,9,14]].copy()

In [82]:
data1.head()

country   incomeperperson alcconsumption lifeexpectancy  \
0  Afghanistan                              .03         48.673   
1      Albania  1914.99655094922           7.29         76.918   
2      Algeria  2231.99333515006            .69         73.131   
3      Andorra  21943.3398976022          10.17                  
4       Angola  1381.00426770244           5.57         51.093   

         employrate  
0  55.7000007629394  
1  51.4000015258789  
2              50.5  
3                    
4  75.6999969482422

# Data Management  

##  changing types

In [83]:
def fun1(a):
    
    for i in range(len(a)):
        
        if a[i] == " ":
            a[i] = np.nan
        else :
            a[i] = float(a[i])
            
        
    

In [84]:
data1.columns = ["country","income","alcohol","life","employment"]

In [85]:
fun1(data1["income"])
fun1(data1["alcohol"])
fun1(data1["life"])
fun1(data1["employment"])

In [86]:
data1.head()

country   income alcohol    life employment
0  Afghanistan      NaN    0.03  48.673       55.7
1      Albania     1915    7.29  76.918       51.4
2      Algeria  2231.99    0.69  73.131       50.5
3      Andorra  21943.3   10.17     NaN        NaN
4       Angola     1381    5.57  51.093       75.7

##  Handling missing values 


In [87]:
def fun2(a):
    x = a
    mn = np.mean(x.dropna())
    for i in range(len(x)):
        if np.isnan(x[i]):
            x[i] = mn
    
    
    


In [88]:
fun2(data1["employment"])
fun2(data1["alcohol"])
fun2(data1["life"])
fun2(data1["income"])

In [89]:
data1.head()

country   income alcohol     life employment
0  Afghanistan  8740.97    0.03   48.673       55.7
1      Albania     1915    7.29   76.918       51.4
2      Algeria  2231.99    0.69   73.131       50.5
3      Andorra  21943.3   10.17  69.7535     58.636
4       Angola     1381    5.57   51.093       75.7

all missing values were replaced by their mean 

#  ANOVA TESTING

IN THIS HYPOTHESIS TESTING I WILL TEST THE RELATION OF INCOME ON ALCOHOL VARIABLES . HERE I WILLL CHANGE THE INCOME VARIABLE INTO CATEGORICAL VARIABLE AND THEN DO THE TEST . 

In [98]:
print(np.percentile(data1["income"],33))
print(np.percentile(data1["income"],66))


1618.088252785783
8740.966076259434


In [99]:
def fun3(a):
    if a<1618.088:
        return 0
    elif a<8740.966:
        return 1
    else :
        return 2

In [100]:
data1["inc"] = list(map(fun3,data1["income"]))

In [101]:
data1.head()

country   income alcohol     life employment  inc
0  Afghanistan  8740.97    0.03   48.673       55.7    2
1      Albania     1915    7.29   76.918       51.4    1
2      Algeria  2231.99    0.69   73.131       50.5    1
3      Andorra  21943.3   10.17  69.7535     58.636    2
4       Angola     1381    5.57   51.093       75.7    0

here you can see three categories were made from income 0 being low 1 being medium and 2 being high

In [102]:
data1["inc"].value_counts()

2    74
0    70
1    69
Name: inc, dtype: int64

each bins have almost same number of distributions

In [117]:
data1["inc"] = data1["inc"].astype("category")

In [118]:
data1.head()

country   income alcohol     life employment inc
0  Afghanistan  8740.97    0.03   48.673       55.7   2
1      Albania     1915    7.29   76.918       51.4   1
2      Algeria  2231.99    0.69   73.131       50.5   1
3      Andorra  21943.3   10.17  69.7535     58.636   2
4       Angola     1381    5.57   51.093       75.7   0

In [124]:
data1["alcohol"] = pd.to_numeric(data1["alcohol"])

## conducting test

In [126]:
model = smf.ols(formula = "alcohol ~ C(inc)" , data = data1).fit().summary()

In [127]:
print(model)

                            OLS Regression Results                            
Dep. Variable:                alcohol   R-squared:                       0.100
Model:                            OLS   Adj. R-squared:                  0.092
Method:                 Least Squares   F-statistic:                     11.72
Date:                Sun, 26 Jul 2020   Prob (F-statistic):           1.49e-05
Time:                        18:10:37   Log-Likelihood:                -615.02
No. Observations:                 213   AIC:                             1236.
Df Residuals:                     210   BIC:                             1246.
Df Model:                           2                                         
Covariance Type:            nonrobust                                         
                  coef    std err          t      P>|t|      [0.025      0.975]
-------------------------------------------------------------------------------
Intercept       4.6184      0.523      8.835      

F - Static is 11.72 , Prob is very low , Hence we can easily reject null hypothesis 

#  post hoc analysis

In [129]:
mc1 = multi.MultiComparison(data1["alcohol"],data1["inc"])
res1 = mc1.tukeyhsd().summary()
print(res1)

Multiple Comparison of Means - Tukey HSD, FWER=0.05
group1 group2 meandiff p-adj  lower  upper  reject
--------------------------------------------------
     0      1   3.1829 0.001  1.4315 4.9342   True
     0      2   2.9933 0.001  1.2719 4.7146   True
     1      2  -0.1896   0.9 -1.9173 1.5382  False
--------------------------------------------------


This showed that 1 and 2 are related while 0 is not related to anyone

#  SUMMARY

These tests showed that high and medium income and their alcohol consumption are related while its not related to low income grps 